src/legacy/core_plugins/kibana/public/management/sections/objects/saved_object_view.tsx

```diff
@@ -0,0 +1,157 @@
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+interface SavedObjectEditionState {
+  type: string;
+  object?: SimpleSavedObject<any>;
+}
+
+export class SavedObjectEdition extends Component<
+  SavedObjectEditionProps,
+  SavedObjectEditionState
+> {
+  constructor(props: SavedObjectEditionProps) {
+    super(props);
+
+    const { serviceRegistry, serviceName } = props;
+    const type = serviceRegistry.get(serviceName)!.service.type;
+
+    this.state = {
+      object: undefined,
+      type,
+    };
+  }
+
+  componentDidMount() {
+    const { id, savedObjectsClient } = this.props;
+    const { type } = this.state;
+    savedObjectsClient.get(type, id).then(object => {
+      this.setState({
+        object,
+      });
+    });
+  }
+
+  render() {
+    const {
+      capabilities,
+      notFoundType,
+      serviceRegistry,
+      id,
+      serviceName,
+      savedObjectsClient,
+    } = this.props;
+    const { type } = this.state;
+    const { object } = this.state;
+    const { edit: canEdit, delete: canDelete } = capabilities.savedObjectsManagement as Record<
+      string,
+      boolean
+    >;
+    const canView = canViewInApp(capabilities, type);
+    const service = serviceRegistry.get(serviceName)!.service;
+
+    return (
+      <div
+        className="kuiViewContent kuiViewContent--constrainedWidth"
+        data-test-subj="savedObjectsEdit"
+      >
+        <Header
+          canEdit={canEdit}
+          canDelete={canDelete}
+          canViewInApp={canView}
+          type={type}
+          onDeleteClick={() => this.delete()}
+          viewUrl={service.urlFor(id)}
+        />
+        {notFoundType && <NotFoundErrors type={notFoundType} />}
+        {canEdit && <Intro />}
+        {object && (
+          <Form
+            object={object}
+            savedObjectsClient={savedObjectsClient}
+            service={service}
+            editionEnabled={canEdit}
+            onSave={this.saveChanges}
+          />
+        )}
+      </div>
+    );
+  }
+
+  async delete() {
+    const { id, savedObjectsClient, overlays, notifications } = this.props;
+    const { type, object } = this.state;
+
+    const confirmed = await overlays.openConfirm(
+      i18n.translate('kbn.management.objects.confirmModalOptions.modalDescription', {
+        defaultMessage: "You can't recover deleted objects",
+      }),
+      {
+        confirmButtonText: i18n.translate(
+          'kbn.management.objects.confirmModalOptions.deleteButtonLabel',
+          {
+            defaultMessage: 'Delete',
+          }
+        ),
+        title: i18n.translate('kbn.management.objects.confirmModalOptions.modalTitle', {
+          defaultMessage: 'Delete saved Kibana object?',
+        }),
+      }
+    );
+    if (confirmed) {
+      await savedObjectsClient.delete(type, id);
+      notifications.toasts.addSuccess(`Deleted '${object!.attributes.title}' ${type} object`);
+      window.location.hash = '/management/kibana/objects';
```

Comment:
Have you looked at using `react-router-dom` to do the redirect? I think it will provide the right interface so that changes to the url structure won't break this.

Reply:
@mattkime I'm planning to do that in the next step, when I actually migrate these two pages to NP plugins (draft in https://github.com/elastic/kibana/pull/59110). ATM, the routes are still provided/handled by angular so imho it doesn't really make sense (if even possible?) to use react router to perform the redirect.